modules/ppcp-button/src/Assets/SmartButton.php

```diff
@@ -354,6 +354,27 @@ private function render_button_wrapper_registrar(): bool {
 
 		add_action( $this->pay_order_renderer_hook(), array( $this, 'button_renderer' ), 10 );
 
+		$not_enabled_on_minicart = $this->settings->has( 'button_mini_cart_enabled' ) &&
+			! $this->settings->get( 'button_mini_cart_enabled' );
+		if (
+		! $not_enabled_on_minicart
+		) {
+			add_action(
+				$this->mini_cart_button_renderer_hook(),
+				static function () {
+                    // phpcs:ignore WordPress.PHP.StrictComparisons.LooseComparison
+					if ( WC()->cart->get_cart_contents_total() == 0 ) {
```

Comment:
@AlexP11223 
I have tried to have a variable `$is_cart_price_total_zero = $this-> is_cart_price_total_zero()` and then `static function () use ($is_cart_price_total_zero )` but it returned wrong value, I guess because it was called outside of the hook. 

In other hands you can't call `$this` inside the handler, right? or is there a better way?
Thanks

Reply:
You can if you either remove `static` here or make `is_cart_price_total_zero` `static`.

Reply:
done 515a06542ed8aa652ac6d0b38b5e68de901f9c73
Thanks.